### PR TITLE
Issue#342 - Charts render as code not charts 

### PR DESCRIPTION
**Pull request by w3creatives**
**Issue Addressed:** 

[342](https://github.com/ytgov/yukon-ca/issues/342) 

**Steps:**

1. Merge the PR
2. Run the **npm run build** command inside **[web-root]/docroot/themes/custom/yukonca_glider**
3. Clear Drupal cache using **drush cr**.

### CODE REVIEW

File: docroot/themes/custom/yukonca_glider/templates/field--node--field-collapsible-items--basic-page.html.twig [hunk 1]
```diff
@@ -38,6 +38,16 @@
   {% if display_items_wrapper_tag ~%}
     <{{ field_items_wrapper_tag|default('div') }}{{ field_items_wrapper_attributes.addClass('field__items') }}>
   {%- endif -%}
+  <div class="yukon-accordion">
+    <div class="yukon-accordion__buttons">
+      <button class="yukon-accordion__expand">
+        <i class="fa fa-plus-square" aria-hidden="true"></i> Expand all
```
Comment (ew-andrew-sus) [inline docroot/themes/custom/yukonca_glider/templates/field--node--field-collapsible-items--basic-page.html.twig:44]: Shouldn't be the text translatable?

File: docroot/themes/custom/yukonca_glider/templates/field--node--field-collapsible-items--basic-page.html.twig [hunk 2]
```diff
@@ -38,6 +38,16 @@
...
+      </button>
+      <button class="yukon-accordion__collapse">
+        <i class="fa fa-minus-square" aria-hidden="true"></i> Collapse all
```
Comment (ew-andrew-sus) [inline docroot/themes/custom/yukonca_glider/templates/field--node--field-collapsible-items--basic-page.html.twig:47]: Shouldn't be the text translatable?

File: docroot/themes/custom/yukonca_glider/src/js/theme.js [hunk 3]
```diff
@@ -39,4 +39,23 @@
       });
     },
   };
+
+  $(document).ready(function () {
```
Comment (ew-andrew-sus) [inline docroot/themes/custom/yukonca_glider/src/js/theme.js:43]: Please use [Behaviors](https://www.drupal.org/node/756722#behaviors) and [once](https://www.drupal.org/node/756722#jquery-once) for custom javascript

File: docroot/themes/custom/yukonca_glider/templates/field--node--field-collapsible-items--basic-page.html.twig (hunk 2 above)
Comment (w3creatives) [reply docroot/themes/custom/yukonca_glider/templates/field--node--field-collapsible-items--basic-page.html.twig]: Done.

File: docroot/themes/custom/yukonca_glider/templates/field--node--field-collapsible-items--basic-page.html.twig (hunk 1 above)
Comment (w3creatives) [reply docroot/themes/custom/yukonca_glider/templates/field--node--field-collapsible-items--basic-page.html.twig]: Done.

File: docroot/themes/custom/yukonca_glider/templates/paragraphs/paragraph--chart--default.html.twig [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+        <div class="field field-name-field-chart-description field-type-text-with-summary field-label-hidden">
+          <div class="field-items">
+            <div class="field-item even"> {{ paragraph.field_description.value|raw }} </div>
+          </div>
+        </div>
+      </div>
+      <section class="yukon-department-sections basic-page-collapsable">
+        <div class="panel-group" id="accordion-{{ paragraph.id() }}" role="tablist">
+          <div class="panel panel-default">
+            <div class="panel-heading" role="tab" id="Employment-insurance-EI-beneficiaries-in-the-Yukon-by-gender">
```
Comment (leeomara) [inline docroot/themes/custom/yukonca_glider/templates/paragraphs/paragraph--chart--default.html.twig:61]: The ID here, `Employment-insurance-EI-beneficiaries-in-the-Yukon-by-gender` looks like it's from one particular chart, not applicable to all charts. Note that this value is used later in the `aria-labelledby` attribue.

**Review by leeomara — COMMENTED** [review]
There are a few places where ID values look suspect.

File: docroot/themes/custom/yukonca_glider/templates/paragraphs/paragraph--chart--default.html.twig [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+        <div class="panel-group" id="accordion-{{ paragraph.id() }}" role="tablist">
+          <div class="panel panel-default">
+            <div class="panel-heading" role="tab">
+              <h4 class="panel-title">
+                <a class="collapsed" data-toggle="collapse" data-parent="#accordion-{{ paragraph.id() }}" href="#accordion-content-{{ paragraph.id() }}" aria-expanded="false" aria-controls="accordion-content-{{ paragraph.id() }}">
+                  {{ 'Source and notes on data'|t }} <span class="float-right text-blue-400 title-icon"><i class="fa fa-plus-square"></i></span>
+                </a>
+              </h4>
+            </div>
+            <div id="accordion-content-{{ paragraph.id() }}" class="panel-collapse collapse" role="tabpanel" aria-labelledby="Employment-insurance-EI-beneficiaries-in-the-Yukon-by-gender" style="">
```
Comment (leeomara) [inline docroot/themes/custom/yukonca_glider/templates/paragraphs/paragraph--chart--default.html.twig:68]: The IDs have been fixed, but the `aria-labelledby` value is still "Employment-insurance-EI-beneficiaries-in-the-Yukon-by-gender". I don't know what the `aria-labelledby` value should be, but I'm pretty sure the current value is wrong.

Comment (w3creatives) [reply docroot/themes/custom/yukonca_glider/templates/paragraphs/paragraph--chart--default.html.twig]: We have cleaned up the html code that we copied from the D7 version. Please check again.

File: docroot/themes/custom/yukonca_glider/templates/paragraphs/paragraph--chart--default.html.twig (hunk 4 above)
Comment (w3creatives) [reply docroot/themes/custom/yukonca_glider/templates/paragraphs/paragraph--chart--default.html.twig]: Done.

File: docroot/themes/custom/yukonca_glider/src/js/theme.js (hunk 3 above)
Comment (w3creatives) [reply docroot/themes/custom/yukonca_glider/src/js/theme.js]: Please check now.

File: docroot/themes/custom/yukonca_glider/templates/paragraphs/paragraph--chart--default.html.twig [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+              </div>
+            </div>
+          </div>
+        </div>
+      </section>
+    </div>
+  </figure>
+</div>
+
+<script>
```
Comment (ew-andrew-sus) [inline docroot/themes/custom/yukonca_glider/templates/paragraphs/paragraph--chart--default.html.twig:84]: Please don't mix Scripts and HTML. Create a separate js file and attach it to this template like `{{ attach_library('mytheme/my-library') }}`
https://www.drupal.org/docs/develop/theming-drupal/adding-assets-css-js-to-a-drupal-theme-via-librariesyml#attach-library-specific-twig

Comment (w3creatives) [reply docroot/themes/custom/yukonca_glider/templates/paragraphs/paragraph--chart--default.html.twig]: It was giving issues getting the dynamic IDs and thats why we took another approach of packing the code in our custom module. Please review and let us know if any changes are required.

**Review by ew-andrew-sus — APPROVED** [review]

**Pull request merged**